src/fund/components/FundCardAmountInput.tsx

```diff
@@ -0,0 +1,167 @@
+import { type ChangeEvent, useCallback, useEffect, useRef } from 'react';
+import { cn, text } from '../../styles/theme';
+import type { FundCardAmountInputPropsReact } from '../types';
+import { formatDecimalInputValue } from '../utils/formatDecimalInputValue';
+import { truncateDecimalPlaces } from '../utils/truncateDecimalPlaces';
+import { FundCardCurrencyLabel } from './FundCardCurrencyLabel';
+
+export const FundCardAmountInput = ({
+  fiatValue,
+  setFiatValue,
+  cryptoValue,
+  setCryptoValue,
+  currencySign,
+  assetSymbol,
+  inputType = 'fiat',
+  exchangeRate = 1,
+}: FundCardAmountInputPropsReact) => {
+  const inputRef = useRef<HTMLInputElement>(null);
+  const hiddenSpanRef = useRef<HTMLSpanElement>(null);
+  const currencySpanRef = useRef<HTMLSpanElement>(null);
+
+  const value = inputType === 'fiat' ? fiatValue : cryptoValue;
+
+  const handleChange = useCallback(
+    (e: ChangeEvent<HTMLInputElement>) => {
+      let value = e.target.value;
+
+      value = formatDecimalInputValue(value);
+
+      if (inputType === 'fiat') {
+        const fiatValue = truncateDecimalPlaces(value, 2);
+        setFiatValue(fiatValue);
+
+        // Calculate the crypto value based on the exchange rate
+        const calculatedCryptoValue = String(
+          Number(value) * Number(exchangeRate),
+        );
+
+        const resultCryptoValue = truncateDecimalPlaces(
+          calculatedCryptoValue,
+          8,
+        );
+        setCryptoValue(calculatedCryptoValue === '0' ? '' : resultCryptoValue);
+      } else {
+        setCryptoValue(value);
+
+        // Calculate the fiat value based on the exchange rate
+        const calculatedFiatValue = String(
+          Number(value) / Number(exchangeRate),
+        );
+        const resultFiatValue = truncateDecimalPlaces(calculatedFiatValue, 2);
+        setFiatValue(resultFiatValue === '0' ? '' : resultFiatValue);
+      }
+    },
+    [exchangeRate, setFiatValue, setCryptoValue, inputType],
+  );
+
+  // biome-ignore lint/correctness/useExhaustiveDependencies: When value changes, we want to update the input width
+  useEffect(() => {
+    if (hiddenSpanRef.current) {
+      const width = Math.max(42, hiddenSpanRef.current.offsetWidth);
+      const currencyWidth =
+        currencySpanRef.current?.getBoundingClientRect().width || 0;
+
+      // Set the input width based on the span width
+      if (inputRef.current) {
+        inputRef.current.style.width = `${width}px`;
+        inputRef.current.style.maxWidth = `${390 - currencyWidth}px`;
```

Comment:
what are you trying to do here?    Something like this may be more useful: https://github.com/coinbase/onchainkit/pull/1224